src/cli/node/json_rpc_client.rs

```diff
@@ -1028,6 +1033,64 @@ pub fn add_peers(addr: SocketAddr, peers: Vec<SocketAddr>) -> Result<(), failure
     Ok(())
 }
 
+#[derive(Serialize, Deserialize)]
+struct SignatureWithData {
+    identifier: String,
+    public_key: String,
+    signature: String,
+}
+
+pub fn claim(
+    addr: SocketAddr,
+    identifier: String,
+    write_to_path: Option<&Path>,
+) -> Result<(), failure::Error> {
+    if identifier.is_empty() || identifier.trim() != identifier {
+        bail!("Claiming identifier cannot be empty or start/end with empty spaces");
+    }
+
+    let request = format!(
+        r#"{{"jsonrpc": "2.0","method": "sign", "params": {:?}, "id": "1"}}"#,
+        calculate_sha256(identifier.as_bytes()).as_ref(),
+    );
+
+    let mut stream = start_client(addr)?;
+    let response = send_request(&mut stream, &request)?;
+
+    let signature: KeyedSignature = parse_response(&response)?;
+    match serde_json::to_string_pretty(&SignatureWithData {
+        identifier: identifier.clone(),
+        public_key: PublicKeyHash::from_public_key(&signature.public_key).to_string(),
```

Comment:
it is a public key or a public key hash?

Reply:
or and address?? xD

Reply:
This is the Display implementation for PublicKeyHash, i.e. a bech32 address